lib/types/library/LibraryFormatter.js

```diff
@@ -276,10 +276,7 @@ class LibraryFormatter extends AbstractUi5Formatter {
 		// Remove base path from fsPath
 		const posixBasePath = this.getSourceBasePath(true);
 
-		// Can match /library/src as well as /library/src/some/namespace
-		const basePathPrefixRegExp = new RegExp(`^${posixBasePath}/?`);
-
-		if (!basePathPrefixRegExp.test(posixFsPath)) {
+		if (!posixFsPath.startsWith(posixBasePath)) {
 			if (posixBasePath === posixFsPath + "/") {
```

Comment:
this condition suggests to me that `posixBasePath` always ends with a slash, but the test mocks the method with a return value without a trailing slash?

Reply:
As discussed we should ensure that `posixBasePath` ends with a slash to make this comparison stable